.github/workflows/preview.yml

```diff
@@ -0,0 +1,99 @@
+name: Preview PR
+
+on:
+  workflow_run:
+    workflows: ["Build and publish"]
+    types: 
+      - completed
+
+jobs:
+# This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-20.04
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+      
+      # Extracts PR from Head SHA
+      - name: Find PR
+        if: ${{ github.event.workflow_run.event == 'pull_request' }}
+        run: |
+          PR=$(curl https://api.github.com/search/issues?q=${{ github.event.workflow_run.head_sha }} |
+          grep -Po "(?<=${{ github.event.workflow_run.repository.full_name }}\/pulls\/)\d*" | head -1)
+          echo "PR=$PR" >> $GITHUB_ENV
+      
+      # Replaces base specs with head specs
+      - name: Pull PR
+        if: ${{ github.event.workflow_run.event == 'pull_request' && env.PR }}
+        run: |
+          rm spec/index.bs
+          rm wgsl/index.bs
+          wget -O spec/index.bs https://raw.githubusercontent.com/${{ github.event.workflow_run.head_repository.full_name }}/${{ github.event.workflow_run.head_sha }}/spec/index.bs
+          wget -O wgsl/index.bs https://raw.githubusercontent.com/${{ github.event.workflow_run.head_repository.full_name }}/${{ github.event.workflow_run.head_sha }}/wgsl/index.bs
+      
+      # Adds Firebase config files to directory
+      - name: Init Firebase
+        if: ${{ github.event.workflow_run.event == 'pull_request' && env.PR }}
+        run: |
+          cat << EOF >> .firebaserc
+          {
+            "projects": {
+              "default": "gpuweb-prs"
+            }
+          }
+          EOF
+          cat << EOF >> firebase.json
+          {
+            "hosting": {
+              "public": "out",
+              "ignore": [
+                "firebase.json",
+                "**/.*",
+                "**/node_modules/**"
+              ]
+            }
+          }
+          EOF
+
+      # Builds Bikeshed specs
+      - name: Build Bikeshed
+        if: ${{ github.event.workflow_run.event == 'pull_request' && env.PR }}
+        run: |
+          git clone --depth 1 https://github.com/tabatkins/bikeshed.git
+          pip3 install --user --editable ./bikeshed
+          export PATH="$(python3 -m site --user-base)/bin:${PATH}"
+          bikeshed update
```

Comment:
I recommend you run `bikeshed update --skip-manifest` here. It'll bypass the manifest repo and get you the most recent information for spec cross references, biblio references, CanIUse, etc. (It takes a bit longer and depends on several other services running, so YMMV. Using the manifest is safer, just may be out of date sometimes.)

If you do, make sure you're running Python 3.7 or 3.8. There's an issue using --skip-manifest under 3.9 (and may be other issues with bikeshed under 3.9...).

Reply:
Great!